neps/nep-0393.md

```diff
@@ -0,0 +1,570 @@
+
+### Example SBT Contract functions
+
+Although the transaction functions below are not part of the SBT smart contract standard (depending on a use case, they may have different parameters), we present here an example interface for SBT issuance and we also provide a reference implementation.
+These functions must relay calls to an SBT registry, which will emit appropriate events.
+We recommend that all functions related to an event will take an optional `memo: Option<String>` argument for accounting purposes.
+
+```rust
+trait SBT {
+    /// the function should overwrite `metadata.class = class`.
+    /// Must provide enough NEAR to cover registry storage cost.
+    // #[payable]
+    fn sbt_mint(
+        &mut self,
+        account: AccountId,
+        metadata: TokenMetadata,
+        memo: Option<String>,
+    ) -> TokenId;
+
+    /// Creates a new, unique token and assigns it to the `receiver`.
+    /// `token_spec` is a vector of pairs: owner AccountId and TokenMetadata.
+    /// Must provide enough NEAR to cover registry storage cost.
+    // #[payable]
+    fn sbt_mint_multi(
+        &mut self,
+        token_spec: Vec<(AccountId, TokenMetadata)>,
+        memo: Option<String>,
+    ) -> Vec<TokenId>;
+
+    // #[payable]
+    fn sbt_recover(&mut self, from: AccountId, to: AccountId, memo: Option<String>);
+
+    fn sbt_renew(&mut self, tokens: Vec<TokenId>, expires_at: u64, memo: Option<String>);
+
+    fn sbt_revoke(token: Vec<u64>, memo: Option<String>) -> bool;
+}
+```
+
+## Reference Implementation
+
+- Common [type definitions](https://github.com/https://github.com/near-ndc/i-am-human/tree/main/contracts/sbt) (events, traits).
+- [I Am Human](https://github.com/https://github.com/near-ndc/i-am-human) registry and issuers.
+
+## Example Flows
+
+In the diagram below we explain a Soul Transfer use case. Alice has two accounts: `alice1` and `alice2`. She is getting tokens from 2 SBT contracts which use the same registry. Later she decides to merge the accounts by doing Soul Transfer.
+
+#### Whitelist a new issuer
+
+An `SBT_1 Issuer` wants to mint tokens using the `SBT_Registry`. Each registry has some way to whitelist an issuer. For this example, let's say that the `SBT_Registry` has a DAO which votes on adding a new issuer:
+
+```mermaid
+sequenceDiagram
+    actor Issuer1 as SBT_1 Issuer
+    actor DAO
+
+    participant SBT_Registry
+
+    Note over Issuer1,DAO: Issuer1 connects with the DAO<br>to be whitelisted.
+
+    Issuer1-->>DAO: request whitelist
+    DAO->>SBT_Registry: whitelist(SBT_1 Issuer)
+```
+
+#### Mint tokens
+
+Alice uses her `alice1` account to interact with `SBT_1 Issuer` and receives an SBT with token ID = 238:
+
+```mermaid
+sequenceDiagram
+    actor Alice
+    actor Issuer1 as SBT_1 Issuer
+
+    participant SBT1 as SBT_1 Contract
+    participant SBT_Registry
+
+    Issuer1->>SBT1: sbt_mint(alice1, class5, metadata)
```

Comment:
@robert-zaremba You upgraded `sbt_mint` prototype, you can remove `class5` from here now...